init.lua

```diff
@@ -5,18 +5,30 @@ sound_api = dofile(modpath .. "/sound_api_core/init.lua")
 minetest.register_on_mods_loaded(function()
     for name, def in pairs(minetest.registered_nodes) do
         if def._sound_def and def._sound_def.key then
-            minetest.override_item(name, {
-                sounds = sound_api[def._sound_def.key](def._sound_def.input)
-            })
+            if sound_api[def._sound_def.key] then
+                minetest.override_item(name, {
+                    sounds = sound_api[def._sound_def.key](def._sound_def.input)
+                })
+            else
+                print("missing sound_api function for sound def key: "..def._sound_def.key)
+                minetest.log("warning", "missing sound_api function for sound def key: "..def._sound_def.key)
+            end
```

Comment:
rather than nesting an if in and if, make it all one line
```lua
if def._sound_def and def._sound_def.key and sound_api[def._sound_def.key] then
    minetest.override_item(name, {
        sounds = sound_api[def._sound_def.key](def._sound_def.input)
    })
else
    print("missing sound_api function for sound def key: "..def._sound_def.key)
    minetest.log("warning", "missing sound_api function for sound def key: "..def._sound_def.key)
end
````

Reply:
I thought about that and decided against it because it will output a warning for all nodes that don't have a `_sound_def`, or a `_sound_def.key`. Maybe the second one could warrant the same warning...

I agree with the function to reduce duplication.

Reply:
ah, right, could change the else to `elseif def._sound_def` to resolve that. see my edit about the function, its kinda pointless, since there both doing different actions, i didnt notice that till i looked at the full code

Reply:
```lua
local modpath = minetest.get_modpath("sound_api")

sound_api = dofile(modpath .. "/sound_api_core/init.lua")

local function validate_sound(key)
    if sound_api[key] then
        return true
    else
        print("attempted to call invalid sound: "..key)
        minetest.log("warning", "attempted to call invalid sound: "..key)
        return false
    end
end

minetest.register_on_mods_loaded(function()
    for name, def in pairs(minetest.registered_nodes) do
        if def._sound_def and def._sound_def.key and validate_sound(def._sound_def.key) then
            minetest.override_item(name, {
                sounds = sound_api[def._sound_def.key](def._sound_def.input)
            })
        end
    end

    local old_reg_node = minetest.register_node
    function minetest.register_node(name, def)
        if def._sound_def and def._sound_def.key and validate_sound(def._sound_def.key) then
            def.sounds = sound_api[def._sound_def.key](def._sound_def.input)
        end

        old_reg_node(name, def)
    end
end)
````
rough suggestion with a common function which sound avoid the above issue